components/brave_wallet_ui/components/market-datatable/index.tsx

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { getLocale } from '../../../common/locale'
+import { BraveWallet, MarketDataTableColumnTypes, SortOrder } from '../../constants/types'
+import { Table, Cell, Header, Row } from '../shared/datatable'
+import {
+  AssetsColumnItemSpacer,
+  AssetsColumnWrapper,
+  StyledWrapper,
+  TableWrapper,
+  TextWrapper
+} from './style'
+import {
+  formatFiatAmountWithCommasAndDecimals,
+  formatPricePercentageChange,
+  formatPriceWithAbbreviation
+} from '../../utils/format-prices'
+import { AssetNameAndIcon } from '../asset-name-and-icon'
+import { AssetPriceChange } from '../asset-price-change'
+import useOnScreen from '../../common/hooks/on-screen'
+import { CoinGeckoText } from '../desktop/views/portfolio/style'
+
+export interface MarketDataHeader extends Header {
+  id: MarketDataTableColumnTypes
+}
+
+export interface Props {
+  headers: MarketDataHeader[]
+  coinMarketData: BraveWallet.CoinMarket[]
+  moreDataAvailable: boolean
+  showEmptyState: boolean
+  onShowMoreCoins: () => void
+  onSort?: (column: MarketDataTableColumnTypes, newSortOrder: SortOrder) => void
+}
+
+const onScreenOptions = {
+  rootMargin: '0px',
+  threshold: 0
+}
+
+const renderCells = (coinMarkDataItem: BraveWallet.CoinMarket) => {
+  const {
+    name,
+    symbol,
+    image,
+    currentPrice,
+    priceChange24h,
+    priceChangePercentage24h,
+    marketCap,
+    marketCapRank,
+    totalVolume
+  } = coinMarkDataItem
+
+  const formattedPrice = formatFiatAmountWithCommasAndDecimals(currentPrice.toString(), 'USD')
```

Comment:
why is USD hard-coded in all of these?

Reply:
@diracdeltas Because the API request to coingecko fetches the data in USD